readme.md

```diff
@@ -130,6 +131,84 @@ console.log(foo());
 
 This means that the unused export `bar` will not be included into the final bundle.
 
+### Tree-Shakeable-Providers
+
+Since the release of Angular 6, The angular team provided a new feature to allow services to be tree-shakeable, meaning that your services will not be included in the final bundle unless they're being used by other services or components. This can help reduce the bundle size by removing unused code from the bundle.
+
+You can make your services tree-shakeable by using the `providedIn` attribute to define where the service should be initialized when using the `@Injectable()` decorator. Then you should remove it from the `providers` attribute of your `NgModule` declaration as well as its import statement as follows.
+
+Before:
+
+app.module.ts
+```
+import { NgModule } from '@angular/core'
+import { AppRoutingModule } from './app-routing.module'
+import { AppComponent } from './app.component'
+import { environment } from '../environments/environment'
+import { MyService } from './app.service'
+
+@NgModule({
+  declarations: [
+    AppComponent
+  ],
+  imports: [
+    ...
+  ],
+  providers: [MyService],
+  bootstrap: [AppComponent]
+})
+export class AppModule { }
+```
+
+my-service.service.ts
+```
```

Comment:
Same as above - move `my-service.service.ts` as a comment in the snippet below and use "```ts"